app/support/products/scheduling_support.rb

```diff
@@ -155,14 +155,19 @@ class ReservationFinder
     attr_accessor :time, :rule, :day_start, :day_end, :options
 
     def initialize(time, rule, options = {})
-      self.time = time
+      self.time = rule.instrument.cutoff_hours > 0 ? time_with_cutoff_hours(time, rule.instrument) : time
```

Comment:
Huh? Yes, because we don't want to try to apply the zero, because we want the time that's passed in if it's zero (if they don't have cutoff hours), not the current time plus 0.